python-checks/src/main/java/org/sonar/python/checks/hotspots/HardCodedCredentialsEntropyCheck.java

```diff
@@ -0,0 +1,214 @@
+      });
+  }
+
+  private void checkRegularArgument(SubscriptionContext subscriptionContext) {
+    var regularArgument = (RegularArgument) subscriptionContext.syntaxNode();
+    var keywordArgument = regularArgument.keywordArgument();
+    if (keywordArgument == null) {
+      return;
+    }
+    var expression = regularArgument.expression();
+    if (!expression.is(Tree.Kind.STRING_LITERAL)) {
+      return;
+    }
+    var value = ((StringLiteral) expression).trimmedQuotesValue();
+    patternMatch(keywordArgument, regularArgument, value, subscriptionContext);
+  }
+
+  private void checkAnnotatedAssignment(SubscriptionContext subscriptionContext) {
+    var annotatedAssignment = (AnnotatedAssignment) subscriptionContext.syntaxNode();
+    var assignedValue = annotatedAssignment.assignedValue();
+    var assignedVariable = annotatedAssignment.variable();
+    if (assignedValue == null || !assignedValue.is(Tree.Kind.STRING_LITERAL) || !assignedVariable.is(Tree.Kind.NAME)) {
+      return;
+    }
+    var value = ((StringLiteral) assignedValue).trimmedQuotesValue();
+    patternMatch(((Name) assignedVariable).name(), assignedValue, value, subscriptionContext);
+  }
+
+  private void checkAssignment(SubscriptionContext subscriptionContext) {
+    var assignment = (AssignmentStatement) subscriptionContext.syntaxNode();
+    var assignedValue = assignment.assignedValue();
+    Map<Expression, Name> expressions = null;
+    if (assignedValue.is(Tree.Kind.TUPLE)) {
+      expressions = Expressions.getExpressionsFromRhs((assignment.assignedValue()))
+        .stream()
+        .collect(HashMap::new, (map, e) -> {
+          var name = Expressions.getAssignedName(e);
+          name.ifPresent(value -> map.put(e, value));
+        }, HashMap::putAll);
+    } else {
+      var name = Expressions.getAssignedName(assignedValue);
+      if (name.isPresent()) {
+        expressions = Map.of(assignedValue, name.get());
+      }
+    }
+
+    if (expressions == null) {
+      return;
+    }
+
+    expressions.entrySet().stream()
+      .filter(entry -> entry.getKey().is(Tree.Kind.STRING_LITERAL))
+      .forEach(entry -> {
+        var expression = entry.getKey();
+        var name = entry.getValue();
+        var value = ((StringLiteral) expression).trimmedQuotesValue();
+        patternMatch(name, expression, value, subscriptionContext);
+      });
+
+  }
+
+  private static boolean valuePassesPostValidation(String value) {
+    return POSTVALIDATION_PATTERN.matcher(value).matches();
+  }
+
+  private boolean entropyShouldRaise(String value) {
+    return ShannonEntropy.calculate(value) > randomnessSensibility;
+  }
+
+  private Collection<Pattern> patterns() {
+    if (patterns == null) {
+      patterns = Stream.of(secretKeyWords.split(","))
+        .map(word -> Pattern.compile("(" + word + ")", Pattern.CASE_INSENSITIVE))
+        .toList();
+    }
+    return patterns;
+  }
+
+  protected static final class ShannonEntropy {
```

Comment:
what is the practical reason for having this class? Why `calculate` could not be just a method of the `Check` itself?

Reply:
The final goal is to put this class in analyzer commons, it is really just a placeholder